core/client/src/light/fetcher.rs

```diff
@@ -489,13 +488,13 @@ pub mod tests {
 
 	#[test]
 	fn storage_read_proof_is_generated_and_checked() {
-		let (local_checker, remote_block_header, remote_read_proof, authorities_len) = prepare_for_read_proof_check();
+		let (local_checker, remote_block_header, remote_read_proof) = prepare_for_read_proof_check();
 		assert_eq!((&local_checker as &FetchChecker<Block>).check_read_proof(&RemoteReadRequest::<Header> {
 			block: remote_block_header.hash(),
 			header: remote_block_header,
 			key: well_known_keys::AUTHORITY_COUNT.to_vec(),
 			retry_count: None,
-		}, remote_read_proof).unwrap().unwrap()[0], authorities_len as u8);
+		}, remote_read_proof).unwrap().unwrap()[0], 3 /*TODO: FIXME*/);
```

Comment:
I'm not sure how to fix this test properly.

Reply:
@svyatonik any ideas?

Reply:
Are authorities still in the storage under `well_known_keys::AUTHORITY_COUNT` key? If so, why change anything here? Alternative option (if you want to get rid of any authorities mentions here, even in tests) - is to query for any storage key value.

Reply:
@svyatonik Could you please elaborate?  Previously `authorities_len` was obtained in the `prepare_for_read_proof_check` function by `remote_client.authorities_at()` call, but I cannot use it anymore.
Should I replace `well_known_keys::AUTHORITY_COUNT` by something else?

Reply:
Something like that should work:
```
let count: u32 = genesis_storage
    .get(well_known_keys::AUTHORITY_COUNT)
    .and_then(|v| Decode::decode(&mut &v[..])).unwrap();
```